go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1249,6 +1255,47 @@ func (call *builtinLastDay) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Date, Flag: arg.Flag | flagNullable}, nil
 }
 
+func (b *builtinFromDays) eval(env *ExpressionEnv) (eval, error) {
+	arg, err := b.arg1(env)
+	if arg == nil || err != nil {
+		return nil, nil
+	}
+
+	days := evalToInt64(arg).i
+	y, m, d := mysqldt.MysqlDateFromDayNumber(int(days))
+
+	// mysql returns 0000-00-00 for days below 366 and above 3652499
+	if y == 0 && m == 0 && d == 0 {
+		return nil, nil
+	}
+
+	// mysql returns NULL if y is greater than 9999
+	if y > 9999 {
+		return nil, nil
+	}
+
+	dt := datetime.NewDateFromStd(time.Date(int(y), time.Month(m), int(d), 0, 0, 0, 0, env.currentTimezone()))
```

Comment:
This conversion seems unneeded? I think you can just straight set the properties from what `MysqlDateFromDayNumber` returns? 

```go
var dt datetime.Date
dt.Year, dt.Month, dt.Day = mysqldt.MysqlDateFromDayNumber(int(arg.i))
```

Then you also only have to check `dt.Year > 9999`, the zero case also automatically falls out? 

Same for the compiler, no need for the conversion there either through Go's Time library.

Reply:
The struct fields of `datetime.Date` are not exported. So, I think we can't follow this.

Also, don't know why I imported `mysql/datetime` twice. Fixing it.

Reply:
Instead of exporting the existing function, we can also add a new one that returns the date already so you don’t have that issue. 

I think we really should avoid going through the go type if not strictly needed here.

Reply:
got it. done.